# Introduction

In this project, we will be looking at the '3G Public Cellular Mobile Telephone Services - Nationwide Service Coverage' data set available on the Data.gov.sg website (2017). We want to see how we can fit a multivariate time series model on the data, and use our model for forecasting. Here, our data set covers the period between 2013 April and 2016 June for the 3 Singapore telephone companies (telcos), Singtel, M1 and Starhub.

# Exploratory Data Analysis

We would first like to import the data from the `csv` file that can be downloaded from the website.

In [90]:
#Importing libraries that we will use
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.dates

#Importing data from csv file into a pandas dataframe
data = pd.read_csv('Telco.csv')
data.head(10)

month    telco  service_coverage
0  2013-04  Singtel             99.11
1  2013-04       M1             97.93
2  2013-04  Starhub             99.25
3  2013-05  Singtel             99.21
4  2013-05       M1             97.87
5  2013-05  Starhub             99.20
6  2013-06  Singtel             99.03
7  2013-06       M1             98.17
8  2013-06  Starhub             99.40
9  2013-07  Singtel             99.33

For our purposes, we will have to reshape the matrix to an  $n×k$  array, where  $n$  is the number of time periods observed, and  $k$  is the number of variables, in this case 3.

In [44]:
data.describe()

service_coverage
count        117.000000
mean          99.526752
std            0.355913
min           97.870000
25%           99.380000
50%           99.630000
75%           99.730000
max           99.930000

The count of service_coverage is 117, which means there are 39 observations for each telco, which corresponds with the time period we mentioned in the introduction.

In [21]:
#Checking if there are any missing values
data.isnull().any()

month               False
telco               False
service_coverage    False
dtype: bool

Lastly, we note that there is no missing data.

We now wish to visualise our data in a plot. We do this by first converting the month column to datetime, and then using the `pivot` function 

In [91]:
pd.to_datetime(data['month'])
%matplotlib inline
data2 = data.pivot(index='month', values='service_coverage', columns='telco' )
data2['month'] = pd.date_range('2013/4', '2016/7', freq='M')
dates = matplotlib.dates.date2num(data2['month'])
plt.plot_date(dates, data2['service_coverage'])

AttributeError: 'numpy.datetime64' object has no attribute 'toordinal'

In [85]:
data2.plot()

ValueError: ordinal must be >= 1

ValueError: ordinal must be >= 1